### PR TITLE
Make taxes work

### LINKED ISSUE #69: [Feature Request]
The tax function currently does nothing in 0.12.1
The expected behavior would be that the sold items and their tax rates are recorded in the memo.

Furthermore, it would be nice to have a toggle button to either include the vat in the amount as it is currently done or add it on top.

### PR DESCRIPTION
**Pull request by talvasconcelos**
First implementation of making taxes work for the tpos.

It's only implemented for the Items stuff, needs integration with normal TPoS usage, with the default tax value.

Big code overhaul (separate js from html) as files were getting massive. Added some components aswell!

There's now a new dropdown for tax related stuff:
image

Taxes can be set per product, if not set, it will take the default value. Tax can be included on price or added on top (inclusive or exclusive).

When not using the TPoS with a product list, the tax is assumed included on price. The cart will display the values of the subtotal / tax amount / total:

tax exclusive:
image

tax inclusive:
image

**Test:**
1. Create a new TPoS, or edit an existing one
2. Change tax settings
3. Use TPoS with the products, pay invoice
4. Switch between tax included or excluded
5. Add different tax value for different products

Closes #69

### CODE REVIEW

File: templates/tpos/tpos.html [hunk 1]
```diff
@@ -24,7 +24,12 @@
     <q-page-sticky v-if="exchangeRate && showPoS" expand position="top">
       <div class="row justify-center full-width">
         <div class="col-12 col-sm-8 col-md-6 col-lg-4 text-center">
-          <h3 class="q-mb-md">${ amountFormatted }</h3>
+          <h3 class="q-mb-md">
+            ${ amountFormatted }
+            <!-- <div class="text-caption" v-if="!taxInclusive">
+              <i>(tax excl.)</i>
+            </div> -->
```
Comment (motorina0) [inline templates/tpos/tpos.html:31]: dead code

**Comment by motorina0** [comment]
@talvasconcelos why is there a tax also on the TPOS level (not only on the product)

**Comment by motorina0** [comment]
This is an important change, some test (unit or Jmeter) should be set in place.

File: migrations.py [hunk 2]
```diff
@@ -108,10 +108,23 @@ async def m007_atm_premium(db):
...
+
+
+async def m009_tax_inclusive(db):
+    """
+    Add tax_inclusive column
+    """
+    await db.execute(
+        "ALTER TABLE tpos.pos ADD COLUMN tax_inclusive BOOL NOT NULL DEFAULT true;"
+    )
+    await db.execute("ALTER TABLE tpos.pos ADD COLUMN tax_default FLOAT;")
```
[the remark this answers is not in the document]
Comment (talvasconcelos) [reply migrations.py:130]: the `tax_default` column?

**Comment by talvasconcelos** [comment]
> Is `Inclusive`/`Exclusive` the English term that businesses use?

Apparently ...

https://x-series-support.lightspeedhq.com/hc/en-us/articles/25533988515355-What-does-tax-inclusive-and-tax-exclusive-mean

**Review by motorina0 — APPROVED** [review]

**Comment by talvasconcelos** [comment]
Can I merge this one, @motorina0 ?

@dni @arcbtc ?

**Review by dni — APPROVED** [review]

**Pull request merged**